Configurable support for T-Mail File Boxes flavours

dolikdev opened the pull request:
Added configurable support for T-Mail File Boxes flavours
compatible to BinkD implementation. Feature is enabled by
default and may be disabled by passing --disable-filebox-flavour
to the configure script.

dolikdev committed:
Added configurable support for T-Mail File Boxes flavours
compatible to BinkD implementation. Feature is enabled by
default and may be disabled by passing --disable-filebox-flavour
to the configure script.

ykaliuta commented:
Thank you a lot for the contribution!

Could it be possible to change the patch a bit:

1) reword the commit message: make a header (first line separated) mentioning part of the functionality (tick) and do it in imperative way;
2) I understsand that the original code is spaghetti, but for me it still looks better to use a helper to both not "repeat yourself" and isolate conditional compilation (but see note below);
3) there is streq() macro which makes code more readable due to absence of reverse condition;
4) may be it makes sense to strcat() the suffix after the base prepared? It makes obvious what is changed in the string so no need to parse printf format and arguments on reading the code;
5) I do not see any reason why it must be compile time configuration. Would the runtime option be better? 

Thanks!

dolikdev committed:
tick: added fileboxes flavours support

Passthrought file boxes directory name will include Hold flavour
attribute (.H) when BoxesWithFlavours is defined in fidogate.conf
and Tick flavour set to hold in routing.

Signed-off-by: Andriy Dzedolik <dz@dolik.dev>

ykaliuta commented:
Pretty much it (I'll most probably reword the commit message a little), but I do not understand why you did not update the patch but made a follow up.

dolikdev commented:
Revoking.
